openadapt/ml/huggingface_models/generic_provider.py

```diff
@@ -0,0 +1,58 @@
+from base_provider import Availability, CompletionProvider, Modality
+from transformers import AutoTokenizer, AutoModel, pipeline
+
+
+class GenericHuggingFaceProvider(CompletionProvider):
+    Name = "Generic HuggingFace Provider"
+    Modalities = [Modality.TEXT]
+    Availabilities = [Availability.HOSTED]
+
+    def create_tokenizer(model_name: str):
+        """
+        Fetches and returns the model's corresponding tokenizer object
+        from HuggingFace.
+        """
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+        return tokenizer
+
+    def fetch_model(model_name: str):
+        """
+        Fetches and returns a model object from HuggingFace
+        """
+        model = AutoModel.from_pretrained(model_name)
+        return model
+
+    def create_pipeline(model_name: str, task: str, trust_remote_code=False):
```

Comment:
make private (add underscore)

Reply:
Decided to remove this completely, unnecessary for now. 

Updated infer method's docstring to contain relevant info about pipeline params.